validate.go

```diff
@@ -57,20 +57,20 @@ func validateRecurse(ast Expression, fields FieldConfigurations, maxRawValueLeng
 		configs := fields
 		var lastConfig *FieldConfiguration
 		// validate the selector
-		for idx, field := range node.Selector {
+		for idx, field := range node.Selector.Path {
 			if fcfg, ok := configs[FieldName(field)]; ok {
 				lastConfig = fcfg
 				configs = fcfg.SubFields
 			} else if fcfg, ok := configs[FieldNameAny]; ok {
 				lastConfig = fcfg
 				configs = fcfg.SubFields
 			} else {
-				return 1, fmt.Errorf("Selector %q is not valid", node.Selector[:idx+1])
+				return 1, fmt.Errorf("Selector %q is not valid", node.Selector.Path[:idx+1])
 			}
 
 			// this just verifies that the FieldConfigurations we are using was created properly
 			if lastConfig == nil {
-				return 1, fmt.Errorf("FieldConfiguration for selector %q is nil", node.Selector[:idx])
+				return 1, fmt.Errorf("FieldConfiguration for selector %q is nil", node.Selector.Path[:idx])
```

Comment:
Ditto here about the error message formatting.

Reply:
Same comment